doc/dfu/quark_dfu_uart.rst

```diff
@@ -0,0 +1,140 @@
+Quark DFU over UART Guide
+#########################
+
+.. contents::
+
+Hardware Setup
+**************
+
+The hardware setup consists of two main steps:
+
+ * Connecting the board to a Linux host using UART.
+ * Having the possibility to connect the DM GPIO pin to GND or to press the DM
+   Button.
+
+The DM GPIO pin is the pin used to put the device into DM mode. The device has
+to be reset while the GPIO is connected to GND. On Quark Developer Boards a
+button connects the DM GPIO to GND if pressed.
+
+Software Setup
+**************
+
+Enabling DM functionality in ROM
+================================
+
+In order to use the Device Management (DM) functionality over UART on the target
+platform, you must install a ROM with such functionality enabled. To do that,
+perform the following steps:
+
+ * Compile the ROM code using the DM_ENABLE flag:
+   ::
+
+       make rom ENABLE_DM=1
+
+ * Flash the new ROM image to the target. (A guide can be found in the
+   main `README.rst <../../README.rst>`__)
+
+Compiling dfu-util-qda host tool
+================================
+
+dfu-util-qda is the tool needed to flash QFU images to the target or manage the
+device.
+
+ * Clone the qm-dfu-util repository
+   ::
+
+        git clone https://github.com/quark-mcu/qm-dfu-util.git
+
+ * Ensure `dh-autoreconf` is installed.
+ * Configure, build and install the tool
+   ::
+
+       cd qm-dfu-util
+       ./autogen.sh
+       ./configure
+       sudo make install
+
+DM functionality usage
+**********************
+
+Creating and flashing a QFU image
+=================================
+
+The following example assumes that you want to use the Bootloader DM
+functionality to flash the LED-blink (blinky) example app to a Quark SE
+development board.
+
+Simple way (using 'make flash'):
+ * Reset the device while connecting the DM GPIO to GND.
+ * Compile, upload and run the example app:
+   ::
+
+       cd examples/blinky
+       make flash SERIAL_PATH=[path_to_serial_interface]
+
+Step by step process (without make flash):
+ * Go to the example directory:
+   ::
+
+       cd examples/blinky
+
+ * Build the project:
+   ::
+
+       make
+
+ * Create the dfu file:
+   ::
+
+       python ../../tools/sysupdate/qm_make_dfu.py -v ./release/quark_se/x86/bin/blinky.bin -p 1
```

Comment:
Can we just add a note:
- Note: The binary-path may differ if built for Quark SE. 
- Note: ARC binaries need to be downloaded to partition 2. (`-p 2`)